File: lint-arwaky/archive/mcp_main_entry.py
"""Bootstrap entry for auto-linter MCP server."""


def main() -> None:
    """Entry point for the auto-linter MCP server."""
    from .agent.agent_container_registry import get_container
    from .surfaces.mcp_server_handler import McpServerHandlerSurface

    # 1. Retrieve the system-wide Dependency Injection container.
    # Entry points are responsible for the first wire-up.
    container = get_container()

    # 2. Instantiate the MCP surface and pass the container as an Aggregate contract.
    surface = McpServerHandlerSurface()

    # 3. Execute the server loop.
    surface.run_server(container)


if __name__ == "__main__":
    main()

File: lint-arwaky/src-rust/mcp_main_entry.rs
use crate::surfaces::mcp_server_handler::McpServerHandlerSurface;
use crate::agent::dependency_injection_container::DependencyInjectionContainer;
use crate::taxonomy::file_path_vo::DirectoryPath;
use std::sync::Arc;

pub async fn run_server() {
    let container = Arc::new(DependencyInjectionContainer::new(DirectoryPath::new(".")));
    let surface = McpServerHandlerSurface::new();
    surface.run_server(container).await;
}
